### PR TITLE
Candidate

### PR DESCRIPTION
**Pull request by biwiki**
@aregtech would you please check this?

### CODE REVIEW

File: examples/10_locsvchello/res/generate.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+ME=$(realpath $0)
+RES_DIR=$(dirname $ME)
+CMD=${RES_DIR}/../../../tools/generate.sh
+
+$CMD --root /home/biwiki/github/areg-sdk/examples/10_locsvchello --doc res/HelloWorld.siml --target _generated_
```
Comment (biwiki) [inline examples/10_locsvchello/res/generate.sh:7]: Users only need to modify this line for every example project

File: tools/generate.sh [hunk 2]
```diff
@@ -1,18 +1,64 @@
...
+
+helpmsg()
+{
+cat << EOF
+$(basename ${ME}) -r <dir> -d <path> [-t <dir>] [-h]
+    -r, --root   <dir>    Project root directory.
+    -d, --doc    <path>   Path to doc file (.siml), relatve to project directory.
+    -t, --target <dir>    Directory for generated code, relatve to project directory.
+                          Default is $TARGET.
+EOF
```
Comment (biwiki) [inline tools/generate.sh:14]: Would this command usage be fine?

File: tools/generate.sh [hunk 3]
```diff
@@ -1,18 +1,64 @@
...
-export PROJECT_ROOT=$(dirname $TOOLS_PATH)
-export CODE_GEN=share/generated
 export CLASSPATH=${TOOLS_PATH}:${TOOLS_PATH}/codegen.jar
 
-exec java com.aregtech.CMFMain --root=${PROJECT_ROOT} --doc=$1 --target=$CODE_GEN
+echo ================================
+echo Generating code to $TARGET ....
+echo ================================
+
+exec java com.aregtech.CMFMain --root=${PROJECT_ROOT} --doc=${DOC_PATH} --target=$TARGET
```
Comment (biwiki) [inline tools/generate.sh:64]: Please take a look at command

**Review by biwiki — COMMENTED** [review]
@aregtech here are several places that need your attention

**Review by aregtech — APPROVED** [review]
thank you

**Pull request merged**